pkg/metrics/metrics.go

```diff
@@ -0,0 +1,85 @@
+package metrics
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/docker/model-distribution/types"
+	"github.com/docker/model-runner/pkg/logging"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sirupsen/logrus"
+)
+
+type Tracker struct {
+	doNotTrack bool
+	transport  http.RoundTripper
+	log        logging.Logger
+}
+
+type TrackerRoundTripper struct {
+	Transport http.RoundTripper
+}
+
+func (h *TrackerRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	clonedReq := req.Clone(req.Context())
+	ctx, cancel := context.WithTimeout(clonedReq.Context(), 5*time.Second)
+	defer cancel()
+	clonedReq = clonedReq.WithContext(ctx)
+	clonedReq.Header.Set("x-docker-model-runner", "true")
+	return h.Transport.RoundTrip(clonedReq)
+}
+
+func NewTracker(httpClient *http.Client, log logging.Logger) *Tracker {
+	client := *httpClient
+	if client.Transport == nil {
+		client.Transport = http.DefaultTransport
+	}
+
+	if os.Getenv("DEBUG") == "1" {
+		if logger, ok := log.(*logrus.Logger); ok {
+			logger.SetLevel(logrus.DebugLevel)
+		} else if entry, ok := log.(*logrus.Entry); ok {
+			entry.Logger.SetLevel(logrus.DebugLevel)
+		}
+	}
+
+	return &Tracker{
+		doNotTrack: os.Getenv("DO_NOT_TRACK") == "1",
+		transport:  &TrackerRoundTripper{Transport: client.Transport},
+		log:        log,
+	}
+}
+
+func (t *Tracker) TrackModel(model types.Model) {
+	if t.doNotTrack {
+		return
+	}
+
+	go t.trackModel(model)
+}
+
+func (t *Tracker) trackModel(model types.Model) {
+	tags := model.Tags()
+	t.log.Debugln("Tracking model:", tags)
+	if len(tags) == 0 {
+		return
+	}
+	ref, err := name.ParseReference(tags[0])
+	if err != nil {
+		t.log.Errorf("Error parsing reference: %v\n", err)
+		return
+	}
+	if _, err = remote.Head(ref, []remote.Option{
+		remote.WithAuthFromKeychain(authn.DefaultKeychain),
+		remote.WithTransport(t.transport),
+		remote.WithUserAgent("docker-model-runner"),
+	}...); err != nil {
```

Comment:
```suggestion
	if _, err = remote.Head(ref,
		remote.WithAuthFromKeychain(authn.DefaultKeychain),
		remote.WithTransport(t.transport),
		remote.WithUserAgent("docker-model-runner"),
	); err != nil {
```
Can't these just be passed in as args rather than structured / destructured?